src/watertap_contrib/seto/unit_models/surrogate/tests/test_chemical_softening.py

```diff
@@ -0,0 +1,211 @@
+import pytest
+from pyomo.environ import (
+    ConcreteModel,
+    value,
+    assert_optimal_termination,
+    units as pyunits,
+)
+import re
+from pyomo.network import Port
+from idaes.core import FlowsheetBlock, UnitModelCostingBlock
+from watertap_contrib.seto.unit_models.zero_order.chemical_softening_0D import (
+    ChemicalSoftening0D,
+)
+
+# from watertap.property_models.seawater_prop_pack import SeawaterParameterBlock
+# from watertap.property_models.water_prop_pack import WaterParameterBlock
+
+from watertap_contrib.seto.property_models.basic_water_properties import (
+    BasicWaterParameterBlock,
+)
+
+from watertap_contrib.seto.costing import (
+    TreatmentCosting,
+    EnergyCosting,
+    SETOSystemCosting,
+)
+
+from idaes.core.util.testing import initialization_tester
+from idaes.core.solvers import get_solver
+from idaes.core.util.model_statistics import (
+    degrees_of_freedom,
+    number_variables,
+    number_total_constraints,
+    number_unused_variables,
+)
+from idaes.core.util.scaling import (
+    calculate_scaling_factors,
+    unscaled_variables_generator,
+    unscaled_constraints_generator,
+    badly_scaled_var_generator,
+)
+import idaes.logger as idaeslog
+
+# Get default solver for testing
+solver = get_solver()
+
+
+class TestChemSoft:
```

Comment:
Since you have a few different CONFIG arguments for the model (softening type, softening train type), ideally we would test all of them but we definitely want to at least test a few of them.

Reply:
Tested each softening train type. I have two softening trains in the test file.